README.md

```diff
@@ -209,8 +120,258 @@ type Droid implements Character {
+- `$info` A value which holds field-specific information relevant to the current query
+
+Lambda function example:
+
+```php
+function ($rootValue, array $arguments, $context, ResolveInfo$info): string {
+    return getHero($arguments['episode'] ?? null);
+}
+``` 
+
+Resolver class example:
+
+```php
+class QueryResolver implements ResolverInterface
+{
+    public function resolveHero($rootValue, array $arguments, $context, ResolveInfo $info): string
+    {
+        return getHero($arguments['episode'] ?? null);
+    }
+}
+```
+
+#### Variables
+
+You can pass in variables when executing a query by passing them to the `graphql` function.
+
+```php
+$query = '
+query HeroNameQuery($id: ID!) {
+  hero(id: $id) {
+    name
+  }
+}';
+
+$variables = ['id' => '1000'];
+
+$result = graphql($schema, $query, null, null, $variables);
+```
+
+#### Context
+
+In case you need to pass in some important contextual information to your queries you can use the `$contextValues` 
+argument on `graphql` to do so. This data will be passed to all of your resolvers as the `$context` argument.
+
+```php
+$contextValues = [
+    'currentlyLoggedInUser' => $currentlyLoggedInUser,
+];
+
+$result = graphql($schema, $query, null, $contextValues, $variables);
+```
+
+### Scalars
+
+The leaf nodes in a schema are called scalars and each scalar resolves to some concrete data. The built-in, or 
+specified scalars in GraphQL are the following:
+
+- Boolean
+- Float
+- Int
+- ID
+- String
+
+#### Custom scalars
+
+In addition to the specified scalars you can also define your own custom scalars and let your schema know about 
+them by passing them to the `buildSchema` function as part of its `$options` argument. 
+
+Custom Date scalar type example:
+
+```php
+$dateType = newScalarType([
+    'name'         => 'Date',
+    'serialize'    => function (DateTime $value) {
+        return $value->format('Y-m-d');
+    },
+    'parseValue'   => function (DateTime $value) {
+        return $value->format('Y-m-d');
+    },
+    'parseLiteral' => function ($node) {
+        if ($node instanceof StringValueNode) {
+            return new DateTime($node->getValue());
+        }
+        return null;
+    },
+]);
+
+$schema = buildSchema($source, [
+    'Query' => QueryResolver::class,
+    [
+        'customTypes' => [$dateType],
+    ],
+]);
+```
+
+Every scalar has to be coerced, which is done by three different functions. The `serialize` function converts a 
+PHP value into the corresponding output value. The`parseValue` function converts a variable input value into the
+corresponding PHP value and the `parseLiteral` function converts an AST literal into the corresponding PHP value.
+
+### Advanced usage
+
+If you are looking for something that isn't yet covered by this documentation your best bet is to take a look at the 
+[tests](./tests) in this project. You'll be surprised how many examples you'll find there.
+
+### Integration
+
+#### Laravel
```

Comment:
IMO instead of adding this whole section on Laravel here we should make a package for it and simply refer to it here.

Reply:
That could be a good idea, but I thought that is a bit short sighted as it’s just a few lines of code. @Jalle19 do you agree that we can keep this for now and create those packages after we release 1.0?

Reply:
I guess